test/tcp_ip/imports/threadlog.py

```diff
@@ -0,0 +1,66 @@
+# --------------------------------------------------------------------------------------------------------------
+# threadlog
+# XC-HWP/ESW3-Queckenstedt
+# 22.01.2025
+# --------------------------------------------------------------------------------------------------------------
+
+import os, sys, time
+import threading
+
+from PythonExtensionsCollection.String.CString import CString
+from PythonExtensionsCollection.Folder.CFolder import CFolder
+
+# --------------------------------------------------------------------------------------------------------------
+# TM***
+
+class threadlog:
+    """Simple class to write content to thread specific log files
+    """
+   
+    def __init__(self, dest_folder=None):
+        if dest_folder is None:
+            self.__dest_folder = CString.NormalizePath(os.path.dirname(os.path.abspath(__file__)))
+        else:
+            self.__dest_folder = CString.NormalizePath(dest_folder)
+        self.__execution_timestamp = time.strftime('%y-%m-%d_%H-%M-%S')
+        handle_dest_folder = CFolder(self.__dest_folder)
+        handle_dest_folder.Create(bOverwrite=False, bRecursive=True)
+        del handle_dest_folder
+        self.__dict_logfile_handles = {}
+        self.__dict_print_count     = {}
+        print(f"thread log files will be written to '{self.__dest_folder}'")
+
+    def __del__(self):
+        for file_handle in self.__dict_logfile_handles.values():
+            file_handle.close()
+            del file_handle
+        del self.__dict_logfile_handles
+        del self.__dict_print_count
+
+    # --------------------------------------------------------------------------------------------------------------
+
+    def tlog(self, ident=None, message=None):
+        ident = str(ident) # integers are accepted as input, but internally we only work with strings
+        thread_id = threading.get_ident()
+        full_ident = f"{ident}_{thread_id}"
```

Comment:
Hi @HolQue ,

Here need to be a Thread Lock. Otherwise it crashes when another thread calls at the same time tlog.

Than you,
 Thohmas

Reply:
Hi Thomas,

I am not sure if I get your point. The outcome of `tlog` is that something is written to a file. The ID of the thread (`thread_id`) is part of the file name. This is the basic idea of this module. It cannot happen that several threads try to write to the same file in parallel. Therefore I would not expect any access violations. Is a thread lock really required nevertheless?

Reply:
Hmm, thinking, thinking, thinking. I plan to add some more parameters to this function, to have more control. In this context I want to configure that all messages are written into one single file. This requires a thread lock.

I will add this in one of the next pull requests. But the current implementation I expect to be thread save.